### PR TITLE
backend/refactor/remove_join_driver_rc_query

### PR DESCRIPTION
**Pull request by 0utkarsh**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [x] Refactoring
- [ ] Dependency updates

## Description

To be merged with https://github.com/nammayatri/shared-kernel/pull/142

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [x] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -3,7 +3,7 @@
     common.url = "github:nammayatri/common";
 
     # Backend inputs
-    shared-kernel.url = "github:nammayatri/shared-kernel";
+    shared-kernel.url = "github:nammayatri/shared-kernel/382fac5fd91cbc6b928f050285cfeed12177c5c2";
```
Comment (srid) [inline flake.nix:6]: This should be reverted. See https://github.com/nammayatri/nammayatri/tree/main/Backend#updating-flake-inputs

Comment (0utkarsh) [reply flake.nix]: Yup, this file was pushed by mistake, removed

**Review by srid — APPROVED** [review]

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Queries/DriverOnboarding/DriverRCAssociation.hs [hunk 2]
```diff
@@ -52,18 +53,58 @@ findAllByDriverId ::
...
+getRcAssocs ::
+  Transactionable m =>
+  Id Person ->
+  m [DriverRCAssociation]
+getRcAssocs driverId = do
+  Esq.findAll $ do
+    rcAssoc <- from $ table @DriverRCAssociationT
     where_ $
       rcAssoc ^. DriverRCAssociationDriverId ==. val (toKey driverId)
     orderBy [desc $ rcAssoc ^. DriverRCAssociationAssociatedOn]
```
Comment (NikithShetty) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Queries/DriverOnboarding/DriverRCAssociation.hs:106]: Is ordering required here?

**Review by NikithShetty — CHANGES_REQUESTED** [review]

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Queries/DriverOnboarding/DriverRCAssociation.hs (hunk 2 above)
Comment (0utkarsh) [reply Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Queries/DriverOnboarding/DriverRCAssociation.hs:104]: Original query had it

File: Backend/dhall-configs/dev/dynamic-offer-driver-app.dhall [hunk 3]
```diff
@@ -105,7 +105,7 @@ in  { esqDBCfg
     , loggerConfig =
             common.loggerConfig
         //  { logFilePath = "/tmp/dynamic-offer-driver-app.log"
-            , logRawSql = False
+            , logRawSql = True
```
[the remark this answers is not in the document]
Comment (0utkarsh) [reply Backend/dhall-configs/dev/dynamic-offer-driver-app.dhall]: Removed, but why don't we have this as true? I can see this is true in rider-app

**Review by NikithShetty — APPROVED** [review]

**Review by NikithShetty — APPROVED** [review]

**Pull request merged**